src/main/java/com/airlines/dao/AirplaneDaoImpl.java

```diff
@@ -0,0 +1,189 @@
+package com.airlines.dao;
+
+import com.airlines.entity.Airplane;
+import com.airlines.entity.CrewMember;
+import com.airlines.entity.Model;
+import com.airlines.exception.DaoOperationException;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.Date;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AirplaneDaoImpl implements AirplaneDao {
+
+    private static final String INSERT_AIRPLANE_SQL = "INSERT INTO airplanes(code_name, model, manufacture, capacity, flight, crew_id) VALUES (?, ?, ?, ?, ?, ?);";
+    private final static String SELECT_AIRPLANE_BY_ID_SQL = "SELECT * FROM airplanes WHERE airplanes.code_name = ?;";
+    private final static String SELECT_ALL_AIRPLANE_SQL = "SELECT * FROM airplanes;";
+    private static final String DELETE_AIRPLANE_SQL = "delete from airplanes where id = ?;";
+    private static final String SELECT_CREW_MEMBERS_BY_CREW_NAME = "SELECT * FROM airplanes LEFT JOIN crews " +
+            "ON airplanes.crews_id = crews.id WHERE crews.name = ?";
+    private static final String UPDATE_AIRPLANE_CREWS_SQL = "UPDATE airplanes SET crew_id = ? WHERE id = ?;";
+
+    private DataSource dataSource;
+
+    @Override
+    public void save(Airplane airplane) {
+        try (Connection connection = dataSource.getConnection()) {
+            saveAirplane(airplane, connection);
+        } catch (SQLException e) {
+            throw new DaoOperationException(e.getMessage(), e);
+        }
+    }
+
+    private void saveAirplane(Airplane airplane, Connection connection) throws SQLException {
+        PreparedStatement insertStatement = prepareInsertStatement(connection, airplane);
+        executeUpdate(insertStatement, "Airplane was not created");
+        int id = fetchGeneratedId(insertStatement);
+        CrewMember.builder().withId(id).build();
+    }
+
+    private PreparedStatement prepareInsertStatement(Connection connection, Airplane airplane) {
+        try {
+            PreparedStatement insertStatement = connection.prepareStatement(INSERT_AIRPLANE_SQL, PreparedStatement.RETURN_GENERATED_KEYS);
+            return fillStatementWithAccountData(insertStatement, airplane);
+        } catch (SQLException e) {
+            throw new DaoOperationException("Cannot prepare statement to insert Airplane", e);
+        }
+    }
+
+    private PreparedStatement fillStatementWithAccountData(PreparedStatement insertStatement, Airplane airplane)
+            throws SQLException {
+        insertStatement.setString(1, airplane.getCodeName());
+        insertStatement.setString(2, airplane.getCodeName());
```

Comment:
please carefully check:
insertStatement.setString(1, airplane.getCodeName());
insertStatement.setString(2, airplane.getCodeName());
and :
   insertStatement.setInt(5, airplane.getCapacity());
        insertStatement.setInt(5, airplane.getFlightRange());
        insertStatement.setInt(5, airplane.getCrews_id());
        
        
        when 5, 5 and 5 duplicating, so:
        
        check all numbers: should be unique, 
        check all fieldNames, i.e. airplane.getCrews_id, should be getCrewId

Reply:
fixed in [3d2fa27](https://github.com/vitaly-lab/airlines-app/pull/6/commits/3d2fa27df397dda1795b8b92332e2576be3067c6)